app/src/main/kotlin/uk/govuk/app/AppViewModel.kt

```diff
@@ -54,6 +56,8 @@ internal class AppViewModel @Inject constructor(
                             shouldDisplayTopicSelection = flagRepo.isTopicsEnabled()
                                     && !appRepo.isTopicSelectionCompleted()
                                     && topicsInitSuccess,
+                            shouldDisplayNotificationsPermission = flagRepo.isNotificationsEnabled()
+                                    && !notificationsClient.permissionDetermined(),
```

Comment:
Everything in `AppUiState.Default()` is starting to get tricky to read - feels like it needs a refactor to mask the logic.

Reply:
Ah, I went through the commits in order. Probs should just check out the change file list. Old GDS habit.